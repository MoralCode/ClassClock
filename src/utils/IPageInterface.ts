export default interface IPageInterface {
    dispatch: any;
}
