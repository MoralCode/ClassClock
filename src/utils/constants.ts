
export const pages = {
    main: "/",
    fullSchedule: "/schedule",
    settings: "/settings",
    fullScheduleLegacy: "/schedule.html",
    settingsLegacy: "/settings.html",
    selectSchool: "/select",
    loginCallback: "/admin/callback",
    welcome: "/welcome",
    admin: "/admin",
    adminBellSchedule: "/admin/bellschedule"
};

export const URLs = {
    github: "https://github.com/MoralCode/ClassClock",
    slack:
        "https://join.slack.com/t/classclock/shared_invite/enQtNTE0MDkyNzAwNzU3LWNhMGUwODU2ZjhkYTYxMTgzNDE1OWEyMGY2OGNiNTBhOWM5NDVhZGUzNDVlNzRiZTE3NTNmODFjYWNkNDhmMDU",
    twitter: "https://twitter.com/classclockapp",
    instagram: "https://www.instagram.com/classclockapp/",
    bluesky: "https://bsky.app/profile/classclockapp.bsky.social",
    discord: "https://discord.classclock.app"
};
