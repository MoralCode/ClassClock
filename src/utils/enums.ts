export enum TimeStates {
    DAY_OFF = "day off",
    OUTSIDE_SCHOOL_HOURS = "outside school hours",
    SCHOOL_IN_CLASS_OUT = "school is in session, but class is not",
    CLASS_IN_SESSION = "class is in session"
}

export enum TimeComparisons {
    IS_BEFORE = -1, //"before",
    IS_DURING_OR_EXACTLY = 0, //"during/exactly",
    IS_AFTER = 1 //"after"
}
