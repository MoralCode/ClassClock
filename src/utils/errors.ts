// errors.ts
// This file contains classes that represent error conditions
// @author Adrian Edwards



/**
 * This custom Error class is intened to be used for passing a minimum wait 
 * time to the .catch() handlers responsible for retrying requests so requests 
 * arent made during a rate-limit coolown period
 * 
 * This class was created with the help of https://javascript.info/custom-errors
 *
 * @export
 * @class RateLimitError
 * @extends {Error}
 */
export class RateLimitError extends Error {
	wait: number;
	constructor(message:string, msToWait: number) {
		super(message);
		this.name = "RateLimitError";
		this.wait = msToWait;
	}
}