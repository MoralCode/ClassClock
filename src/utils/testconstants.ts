// This is not actually a unit test, just a central place to hold frequently-used constants that are often reused across tests

import Time from "../@types/time";
import School from "../@types/school";
import BellSchedule from "../@types/bellschedule";
import ClassPeriod from "../@types/classperiod";





export const currentDate = new Date("2019-07-28T07:37:50.634Z");

export const className = "First Period";

export const beforeClass = new Time(8, 0);
export const beforeClassString = "8:00"
export const startTime = new Time(8, 25);
export const startTimeString = "8:25";
export const classDuration = new Time(1, 30);
export const duringClass = new Time(9, 0);
export const duringClassString = "9:00";
export const endTime = new Time(9, 55);
export const endTimeString = "9:55";
export const afterClass = new Time(10, 0);
export const afterClassString = "10:00";

export const classPeriod = new ClassPeriod(className, startTime, endTime, currentDate);

export const classPeriodJSON = {
           name: className,
           startTime: startTimeString,
           endTime: endTimeString,
           creationDate: currentDate
       };

export const bellScheduleId = "1";
export const bellScheduleName = "Regular Schedule";
export const bellScheduleEndpoint = "/path/to/sched";
export const bellScheduleDisplayName = "Display Name";


export const bellSchedule = new BellSchedule(
    bellScheduleId,
    bellScheduleName,
    bellScheduleEndpoint,
    [
        new Date("2019-07-28T07:37:50.634Z"),
        new Date("2019-07-29T07:38:10.979Z"),
        new Date("2019-07-23T07:38:28.263Z")
    ],
    [classPeriod],
    currentDate
);

export const bellScheduleJSON = {
           id: bellScheduleId,
           name: bellScheduleName,
           endpoint: bellScheduleEndpoint,
           dates: [
               "2019-07-28T07:37:50.634Z",
               "2019-07-29T07:38:10.979Z",
               "2019-07-23T07:38:28.263Z"
           ],
           classes: [classPeriodJSON],
           lastModified: currentDate
       };


export const schoolId = "10";
export const schoolName = "The High School";
export const schoolAcronym = "THS";
export const schoolEndpoint = "/some/endpoint";
export const schoolTimezone = "America/Cupertino";
export const passingPeriodName = "Transition Time";

export const school = new School(
    schoolId,
    schoolName,
    schoolAcronym,
    schoolEndpoint,
    schoolTimezone,
    [bellSchedule],
    passingPeriodName,
    currentDate,
    currentDate
);

export const schoolJSON = {
           id: schoolId,
           name: schoolName,
           acronym: schoolAcronym,
           endpoint: schoolEndpoint,
           timezone: schoolTimezone,
           schedules: [bellScheduleJSON],
           passingPeriodName,
           creationDate: currentDate,
           lastModified: currentDate
       };
