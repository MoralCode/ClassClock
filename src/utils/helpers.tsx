import Time from "../@types/time";
import School from "../@types/school";
import { TimeComparisons, TimeStates } from "./enums";
import ClassPeriod from "../@types/classperiod";

export function getValueIfKeyInList(list: string[], object: any) {
    for (const key of list) {
        if (object.hasOwnProperty(key)) {
            return object[key];
        }
    }
}

export function deconstructJsonApiResource(json: any) {
    const data = {
        type: json.type,
        id: json.id,
        ...(json.links !== undefined && {
            endpoint: json.links.self
        })
    };
    return Object.assign({}, data, json.attributes);
}

export function getCurrentDate() {
    return new Date();
}

/**
 * @returns a flag that represents the current chunk of time categorically
 */
export function getTimeStateForDateAtSchool(date: Date, school: School) {
    const currentBellSchedule = school.getScheduleForDate(date);

    //there is no schedule that applies today
    if (!currentBellSchedule) {
        return TimeStates.DAY_OFF;
    }

    const currentClassPeriod = currentBellSchedule.getClassPeriodForTime(
        Time.fromDate(date)
    );

    //it is a school day but it is not school hours
    if (!school.isInSession(date)) {
        return TimeStates.OUTSIDE_SCHOOL_HOURS;
    }

    //the current time lies between the start of the first schedules class and the end of the last
    else if (school.isInSession(date) && !currentClassPeriod) {
        return TimeStates.SCHOOL_IN_CLASS_OUT;
    }

    //the current time lies within a scheduled class period
    else if (currentClassPeriod !== undefined) {
        return TimeStates.CLASS_IN_SESSION;
    }
}

/**
 * @returns the next relevent time to count down to
 */
export function getNextImportantTime(
    date: Date,
    school: School
): [ClassPeriod, Time] | undefined {
    const currentBellSchedule = school.getScheduleForDate(date);

    //there is no schedule that applies today
    if (!currentBellSchedule) {
        return;
    }

    currentBellSchedule.getAllClasses().forEach((value: ClassPeriod) => {
        for (const time of [value.getStartTime(), value.getEndTime()]) {
            if (time.getMillisecondsTo(Time.fromDate(date)) >= 0) {
                return [value, time];
            }
        }
    });
}

/**
 * This export function checks if the current time is between the two given times
 * This is useful for checking which class period you are currently in or for checking if school is in session.
 *
 * @param {*} checkTime the time that the check results are returned for
 * @param {*} startTime the start time of the range to check
 * @param {*} endTimethe the end time of the range to check
 *
 * @returns -1 if checkTime is before range, 0 if checkTime is within range, 1 if checkTime is after range
 */
export function checkTimeRange(checkTime: Time, startTime: Time, endTime: Time) {
    if (startTime.getMillisecondsTo(endTime) <= 0) {
        //theres a problem
    }
    const startCheck = checkTime.getMillisecondsTo(startTime);
    const endCheck = checkTime.getMillisecondsTo(endTime);

    if (startCheck > 0) {
        return TimeComparisons.IS_BEFORE;
    } else if (endCheck < 0) {
        return TimeComparisons.IS_AFTER;
    } else {
        return TimeComparisons.IS_DURING_OR_EXACTLY;
    }
}
