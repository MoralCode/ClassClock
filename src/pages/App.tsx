import React, { Component, useEffect, useState } from "react";
import { connect } from "react-redux";
import { push } from "redux-first-routing";
import "../global.css";
import Link from "../components/Link";
import Block from "../components/Block/Block";
import { DateTime } from "luxon";
import School from "../@types/school";
import { pages } from "../utils/constants";
import { ISchoolsState, SelectedSchoolState } from "../store/schools/types";
import { getCurrentDate } from "../utils/helpers";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCog } from "@fortawesome/free-solid-svg-icons";
import { ISettingsState, IUserSettings } from "../store/usersettings/types";
import StatusIndicator from "../components/StatusIndicator";
import ClassClockService from "../services/classclock";
import { selectSchool } from "../store/schools/actions";
import ClassPeriod from "../@types/classperiod";
import Time from "../@types/time";

export interface IAppProps {
    selectedSchool: SelectedSchoolState;
    userSettings: IUserSettings;
    error: string;
    dispatch: any;
}

export const App = (props: IAppProps) => {
    const [currentDate, setDate] = useState(getCurrentDate());
    const [online, setOnline] = useState(true);
    const [connected, setConnected] = useState(false);

    const navigate = (to: string) => {
        props.dispatch(push(to));
    };

    const updateConnectionState = () => {
        ClassClockService.isReachable().then((reachable) => {
            setConnected(reachable)
        })
    }

    useEffect(() => {
        const timingInterval: NodeJS.Timeout = setInterval(() => {
            setDate(getCurrentDate());
        }, 500);

        //set the connected state immediately on pageload
        updateConnectionState()

        //then schedule the connection state to be updated every 2 min
        const connectivityInterval: NodeJS.Timeout = setInterval(() => {
            updateConnectionState()
        }, 120000);

        //check when the schedule was last updated
        const dataAge = DateTime.local().toMillis() - props.selectedSchool.lastUpdated
                
        // if data is > 12 hours old 43200000
        if (dataAge > 43200000){
            props.dispatch(selectSchool(props.selectedSchool.data.getIdentifier()))
        }

        return () => {
            clearInterval(timingInterval)
            clearInterval(connectivityInterval)
        };
    }, []);

    window.addEventListener('online', () => {
        setOnline(true)
        updateConnectionState()
    });
    window.addEventListener('offline', () => {setOnline(false)});

    //If this is not present here, then there will be an error when the next lines run in the event data is being fetched and there is no stored school.
    // this guards against trying to call a function on an empty data object which may be present if the school is being fetched for the first time.
    if (props.selectedSchool.isFetching) {
        return <p>Fetching...</p>
    }
    const currentSchedule = props.selectedSchool.data.getScheduleForDate(currentDate);
    const schoolTimezone = props.selectedSchool.data.getTimezone();

    const getContent = () => {
        switch (currentSchedule) {
            case undefined:
                if (!props.selectedSchool.isFetching) {
                    props.dispatch(push(pages.welcome));
                }
                return
            case null:
                return <p>No School Today</p>;
            default:

                let nextClass: ClassPeriod | undefined = currentSchedule.getClassStartingAfter(currentDate, schoolTimezone);
                let nextImportantTime: Time | undefined;

                const currentClass = currentSchedule.getClassPeriodForTime(currentDate, schoolTimezone);

                if (currentClass){
                    nextImportantTime = currentClass.getEndTime()
                } else if (nextClass) {
                    nextImportantTime = nextClass.getStartTime()
                } else {
                    return <p>School's Out!</p>;
                }

                return (
                    <>
                    <Block>
                        <p>
                            Today is a{" "}
                            <Link
                                // tslint:disable-next-line: jsx-no-lambda
                                destination={() => navigate(pages.fullSchedule)}
                                id="viewScheduleLink"
                            >
                                {currentSchedule.getName()}
                            </Link>
                        </p>
                    </Block>
                    <Block>
                        <p>You are currently in: </p>
                        <p className="timeFont" style={{ fontSize: "30px" }}>
                            <b>
                                {currentClass !== undefined
                                    ? currentClass.getName()
                                    : props.selectedSchool.data.getPassingTimeName()}
                            </b>
                        </p>
                    </Block>
                    <Block>
                        <p>...which ends in:</p>
                        {/* <h1 className="centered bottomSpace time bigger" id="timeToEndOfClass" /> */}
                        <p className="timeFont" style={{ fontSize: "60px" }}>
                            <b>
                                {nextImportantTime
                                        ? nextImportantTime.getTimeDeltaTo(Time.fromDateTime(currentDate, props.selectedSchool.data.getTimezone())).getFormattedString(false, true)
                                        : "No Class"}
                            </b>
                        </p>
                        <p>Your next class period is: </p>
                        <p className="timeFont" style={{ fontSize: "30px" }}>
                            <b>{nextClass ? nextClass.getName() : "No Class"}</b>
                        </p>
                    </Block>
                    </>
                );
        }
    }

    const getStatus = () => {
        let content: JSX.Element = <></>;
        let color = "";

        if (props.selectedSchool.isFetching){
            color = "yellow";
            content = <>Refreshing...</>
        } else if (props.error != "") {
            color = "red";
            content = <>Error</>
        } else if (!online) {
            color = "orange";
            content = <>Offline</>
        } else {
            content =  connected? <>Connected</> : <>Online</>;
            color = "green";
        }

        return <StatusIndicator color={color}>{content}</StatusIndicator>
    }

    return (
        <>
        <div className="App">
            <Link
                className="cornerNavButton cornerNavTop cornerNavLeft smallIcon"
                // tslint:disable-next-line: jsx-no-lambda
                destination={() => navigate(pages.settings)}
            >
                <FontAwesomeIcon icon={faCog} />
            </Link>
            <br />
            <Block>
                <p>It is currently: </p>
                <p className="timeFont" style={{ fontSize: "40px" }}>
                    {currentDate.toLocaleString({ hour: '2-digit', minute: '2-digit', second: '2-digit', hour12: !props.userSettings.use24HourTime }) }
                </p>
                <p>
                    on{" "}
                    <b>
                        {currentDate.toLocaleString({
                            weekday: "long",
                            year: "numeric",
                            month: "short",
                            day: "numeric"
                        })}
                    </b>
                </p>
            </Block>

            {getContent()}
        </div>
        {getStatus()}
        </>
    );
};

const mapStateToProps = (state: ISchoolsState & ISettingsState & {error: string}) => {
    const { selectedSchool, userSettings, error } = state;
    return { selectedSchool, userSettings, error };
};

export default connect(mapStateToProps)(App);
