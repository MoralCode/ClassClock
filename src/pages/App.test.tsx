import React from 'react';
import ReactDOM from 'react-dom';
import renderer from "react-test-renderer";
import { App } from './App';
import { school, beforeSchoolHours, noSchool, inClass, betweenClass, afterSchoolHours } from '../utils/testconstants';
import MockDate from 'mockdate';

// make a fake dispatch function, it doesnt need to do anything, because we assume that has already been tested in the redux library
const mockDispatch = jest.fn();

const mockState = {
    isFetching: false,
    didInvalidate: false,
    data: school,
    lastUpdated: 1564299400 //Sun Jul 28 2019 00:36:40 UTC-0700 (Pacific Daylight Time)
};

const rawComponent = 
    <App selectedSchool={mockState} dispatch={mockDispatch} userSettings={{ use24HourTime: true }} />



describe("App", () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(rawComponent, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("shows the correct screen before school hours", () => {
      MockDate.set(beforeSchoolHours);
      const tree = renderer.create(rawComponent).toJSON();
      expect(tree).toMatchSnapshot();
  });

  it("shows the correct screen on no school days", () => {
      MockDate.set(noSchool);
      const tree = renderer.create(rawComponent).toJSON();
      expect(tree).toMatchSnapshot();
  });

  it("shows the correct screen when class is in session", () => {
      MockDate.set(inClass);
      const tree = renderer.create(rawComponent).toJSON();
      expect(tree).toMatchSnapshot();
  });

  it("shows the correct screen between class", () => {
      MockDate.set(betweenClass);
      const tree = renderer.create(rawComponent).toJSON();
      expect(tree).toMatchSnapshot();
  });

  it("shows the correct screen after school", () => {
      MockDate.set(afterSchoolHours);
      const tree = renderer.create(rawComponent).toJSON();
      expect(tree).toMatchSnapshot();
  });

});


