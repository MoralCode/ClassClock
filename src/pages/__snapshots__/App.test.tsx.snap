// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App shows the correct screen after school 1`] = `
<div
  className="App"
>
  <a
    className="cornerNavButton cornerNavLeft smallIcon"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-cog fa-w-16 "
      data-icon="cog"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
  <br />
  <div
    className="infoBlock"
  >
    <p>
      It is currently: 
    </p>
    <p
      className="timeFont"
      style={
        Object {
          "fontSize": "40px",
        }
      }
    >
      05:00:50
    </p>
    <p>
      on
       
      <b>
        Sunday, Jul 28, 2019
      </b>
    </p>
  </div>
  <div
    className="infoBlock"
  >
    <p>
      Today is a
       
      <a
        id="viewScheduleLink"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Regular Schedule
      </a>
    </p>
  </div>
  <div
    className="infoBlock"
  >
    <p>
      You are currently in: 
    </p>
    <p
      className="timeFont"
      style={
        Object {
          "fontSize": "30px",
        }
      }
    >
      <b>
        Transition Time
      </b>
    </p>
  </div>
  <div
    className="infoBlock"
  >
    <p>
      ...which ends in:
    </p>
    <p
      className="timeFont"
      style={
        Object {
          "fontSize": "60px",
        }
      }
    >
      <b>
        03:24:10
      </b>
    </p>
    <p>
      Your next class period is: 
    </p>
    <p
      className="timeFont"
      style={
        Object {
          "fontSize": "30px",
        }
      }
    >
      <b>
        First Period
      </b>
    </p>
  </div>
</div>
`;

exports[`App shows the correct screen before school hours 1`] = `
<div
  className="App"
>
  <a
    className="cornerNavButton cornerNavLeft smallIcon"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-cog fa-w-16 "
      data-icon="cog"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
  <br />
  <div
    className="infoBlock"
  >
    <p>
      It is currently: 
    </p>
    <p
      className="timeFont"
      style={
        Object {
          "fontSize": "40px",
        }
      }
    >
      01:00:00
    </p>
    <p>
      on
       
      <b>
        Sunday, Jul 28, 2019
      </b>
    </p>
  </div>
  <div
    className="infoBlock"
  >
    <p>
      Today is a
       
      <a
        id="viewScheduleLink"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Regular Schedule
      </a>
    </p>
  </div>
  <div
    className="infoBlock"
  >
    <p>
      You are currently in: 
    </p>
    <p
      className="timeFont"
      style={
        Object {
          "fontSize": "30px",
        }
      }
    >
      <b>
        Transition Time
      </b>
    </p>
  </div>
  <div
    className="infoBlock"
  >
    <p>
      ...which ends in:
    </p>
    <p
      className="timeFont"
      style={
        Object {
          "fontSize": "60px",
        }
      }
    >
      <b>
        07:25:00
      </b>
    </p>
    <p>
      Your next class period is: 
    </p>
    <p
      className="timeFont"
      style={
        Object {
          "fontSize": "30px",
        }
      }
    >
      <b>
        First Period
      </b>
    </p>
  </div>
</div>
`;

exports[`App shows the correct screen between class 1`] = `
<div
  className="App"
>
  <a
    className="cornerNavButton cornerNavLeft smallIcon"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-cog fa-w-16 "
      data-icon="cog"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
  <br />
  <div
    className="infoBlock"
  >
    <p>
      It is currently: 
    </p>
    <p
      className="timeFont"
      style={
        Object {
          "fontSize": "40px",
        }
      }
    >
      03:00:50
    </p>
    <p>
      on
       
      <b>
        Sunday, Jul 28, 2019
      </b>
    </p>
  </div>
  <div
    className="infoBlock"
  >
    <p>
      Today is a
       
      <a
        id="viewScheduleLink"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Regular Schedule
      </a>
    </p>
  </div>
  <div
    className="infoBlock"
  >
    <p>
      You are currently in: 
    </p>
    <p
      className="timeFont"
      style={
        Object {
          "fontSize": "30px",
        }
      }
    >
      <b>
        Transition Time
      </b>
    </p>
  </div>
  <div
    className="infoBlock"
  >
    <p>
      ...which ends in:
    </p>
    <p
      className="timeFont"
      style={
        Object {
          "fontSize": "60px",
        }
      }
    >
      <b>
        05:24:10
      </b>
    </p>
    <p>
      Your next class period is: 
    </p>
    <p
      className="timeFont"
      style={
        Object {
          "fontSize": "30px",
        }
      }
    >
      <b>
        First Period
      </b>
    </p>
  </div>
</div>
`;

exports[`App shows the correct screen on no school days 1`] = `
<div
  className="App"
>
  <a
    className="cornerNavButton cornerNavLeft smallIcon"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-cog fa-w-16 "
      data-icon="cog"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
  <br />
  <div
    className="infoBlock"
  >
    <p>
      It is currently: 
    </p>
    <p
      className="timeFont"
      style={
        Object {
          "fontSize": "40px",
        }
      }
    >
      00:37:50
    </p>
    <p>
      on
       
      <b>
        Saturday, Jul 27, 2019
      </b>
    </p>
  </div>
  <p>
    No School Today
  </p>
</div>
`;

exports[`App shows the correct screen when class is in session 1`] = `
<div
  className="App"
>
  <a
    className="cornerNavButton cornerNavLeft smallIcon"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-cog fa-w-16 "
      data-icon="cog"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
  <br />
  <div
    className="infoBlock"
  >
    <p>
      It is currently: 
    </p>
    <p
      className="timeFont"
      style={
        Object {
          "fontSize": "40px",
        }
      }
    >
      02:00:50
    </p>
    <p>
      on
       
      <b>
        Sunday, Jul 28, 2019
      </b>
    </p>
  </div>
  <div
    className="infoBlock"
  >
    <p>
      Today is a
       
      <a
        id="viewScheduleLink"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Regular Schedule
      </a>
    </p>
  </div>
  <div
    className="infoBlock"
  >
    <p>
      You are currently in: 
    </p>
    <p
      className="timeFont"
      style={
        Object {
          "fontSize": "30px",
        }
      }
    >
      <b>
        Transition Time
      </b>
    </p>
  </div>
  <div
    className="infoBlock"
  >
    <p>
      ...which ends in:
    </p>
    <p
      className="timeFont"
      style={
        Object {
          "fontSize": "60px",
        }
      }
    >
      <b>
        06:24:10
      </b>
    </p>
    <p>
      Your next class period is: 
    </p>
    <p
      className="timeFont"
      style={
        Object {
          "fontSize": "30px",
        }
      }
    >
      <b>
        First Period
      </b>
    </p>
  </div>
</div>
`;
